File: lecture04/03_contiguous/template.py
#!/usr/bin/env python3

import collections
import sys

# Functions


def find_maximum_contiguous_array(array):
    # TODO: find length of maximum contiguous array
    return 0
        
# Main execution

if __name__ == '__main__':
    for line in sys.stdin:
        digits = map(int, line.split())
        print(find_maximum_contiguous_array(digits))

File: lecture04/03_contiguous/template.cpp
// continuous_array.cpp

#include <iostream>
#include <string>
#include <sstream>
#include <unordered_map>
#include <vector>

using namespace std;

// Functions

vector<bool> read_digits(string &s) {
    vector<bool> digits;
    stringstream ss(s);
    int digit;

    while (ss >> digit) {
	digits.push_back(digit);
    }

    return digits;
}

int find_maximum_contiguous_array(vector<bool> &digits) {
    // TODO: find length of maximum contiguous array
    return 0;
}

// Main execution

int main(int argc, char *argv[]) {
    string line;
    while (getline(cin, line)) {
    	auto digits = read_digits(line);
    	auto length = find_maximum_contiguous_array(digits);
    	cout << length << endl;
    }

    return 0;
}
